File: app/src/main/java/com/example/helloworld/MainActivity.java
package com.example.helloworld;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);



    }
    public void onRegisterBtnClick (View view){
        TextView textViewName = findViewById(R.id.textViewName);
        textViewName.setTextColor(getResources().getColor(R.color.my_purple));
        EditText edtTxtName = findViewById(R.id.edtTxtName);
        textViewName.setText( edtTxtName.getText().toString());

        TextView textViewLastName = findViewById(R.id.textViewLastName);
        textViewName.setTextColor(getResources().getColor(R.color.my_purple));
        EditText edtTxtLastName = findViewById(R.id.edtTxtLastName);
        textViewLastName.setText( edtTxtLastName.getText().toString());

        TextView textViewEmail = findViewById(R.id.textViewEmail);
        textViewName.setTextColor(getResources().getColor(R.color.my_purple));
        EditText edtEmail = findViewById(R.id.edtEmail);
        textViewEmail.setText( edtEmail.getText().toString());
    }


}

File: app/src/main/java/com/example/myapplication/MainActivity.kt
package com.example.myapplication

import android.app.DatePickerDialog
import android.app.TimePickerDialog
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.CountDownTimer
import com.example.myapplication.service.AlarmService
import kotlinx.android.synthetic.main.activity_main.*
import java.util.*

class MainActivity : AppCompatActivity() {
    lateinit var alarmService: AlarmService
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        alarmService = AlarmService(this)

        setExact.setOnClickListener {
            setAlarm { alarmService.setExactAlarm(it) }
        }

        setRepetitive.setOnClickListener { setAlarm { alarmService.setRepetitiveAlarm(it) } }
    }

    private fun setAlarm(callback: (Long) -> Unit) {
        val timer = object: CountDownTimer(20000,10000){
            override fun onTick(millisUntilFinished: Long) {
                Calendar.getInstance().apply {
                    this.set(Calendar.SECOND, 0)
                    this.set(Calendar.MILLISECOND, 0)
                    this.get(Calendar.HOUR)
                    this.get(Calendar.MINUTE)
                    callback(this.timeInMillis)

                }
            }

            override fun onFinish() {
              setAlarm (callback)
            }
        }
        timer.start()
        }

}